### PR TITLE
Add capability to set working directory as a command line argument.

### PR DESCRIPTION
**Pull request by zacharied**
As it stands, running `alacritty -e 'cd foo'` causes `Command::spawn()` to fail with OS error code `2`. With this feature, running `alacritty -c foo` runs alacritty from the working directory `foo` via `libc::chdir`, with the intended purpose of starting the shell from a specific directory. It mirrors the functionality of the `-cd` flag from `urxvt`.

### CODE REVIEW

**Comment by NickeZ** [comment]
Maybe you could use the safe and portable interface instead? https://doc.rust-lang.org/std/env/fn.set_current_dir.html

**Comment by zacharied** [comment]
... didn't even see that. My bad. I'll get to work on that.

File: src/cli.rs [hunk 1]
```diff
@@ -117,6 +124,10 @@ impl Options {
             3 | _ => options.log_level = log::LogLevelFilter::Trace
         }
 
+        if let Some(dir) = matches.value_of("chdir") {
+            options.chdir = Some(PathBuf::from(dir.to_string()));
```
Comment (jwilm) [inline src/cli.rs]: I don't think the to_string call is necessary here.

File: src/cli.rs [hunk 2]
```diff
@@ -75,6 +78,10 @@ impl Options {
                 .multiple(true)
                 .conflicts_with("q")
                 .help("Increases the level of verbosity (the max level is -vvv)"))
+            .arg(Arg::with_name("chdir")
+                 .short("c")
```
Comment (jwilm) [inline src/cli.rs]: Let's drop the short form here

Comment (jwilm) [inline src/cli.rs]: I would like to try and match other terminal emulators with this flag. The standard seems to be `--working-directory`.

**Comment by zacharied** [comment]
This pull request has been fixed with the suggestions of @jwilm. Let me know if there's anything else you want changed!

File: src/tty.rs [hunk 3]
```diff
@@ -235,6 +236,13 @@ pub fn new<T: ToWinsize>(config: &Config, options: &Options, size: T) -> Pty {
         Ok(())
     });
 
+    // Handle set working directory option
+    if let Some(ref dir) = options.working_dir {
+        env::set_current_dir(dir.as_path()).unwrap_or_else(|e| {
```
Comment (jwilm) [inline src/tty.rs]: Why not use [`std::process::Command::current_dir`](https://doc.rust-lang.org/1.2.0/std/process/struct.Command.html#method.current_dir)?

**Comment by zacharied** [comment]
That's a good suggestion. Added.

**Comment by jwilm** [comment]
Beautiful!

Would you mind squashing your commits? Ready to merge after that.

**Comment by jwilm** [comment]
Oh, it looks like the branch needs to be rebased as well. Something that conflicts must have been merged since this was opened. Sorry about that!

I just wanted to share with you that I'm pretty excited about this feature. I'm planning to setup a launcher for i3 which first checks if the active window is Alacritty, reads the process's (or I guess the shell process's) working directory and then spawns a new instance using this flag. :smile:

**Comment by jwilm** [comment]
@zacharied sure!

Squashing is the act of coalescing _n_ commits into a smaller number of commits. In this case, we'd like to just have a single commit for this feature.

Rebasing is the act of changing which commit your branch started from. In this case, you started on _master_ at some point in the past, but the _master_ branch has advanced during the time this PR has been open. Rebasing in this case means moving your commit to "be on top of" the current _master_.

For more info, [GitHub's knowledge base article about rebasing](https://help.github.com/articles/about-git-rebase/) is a good place to start.

It's slightly complicated by the fact that you've developed directly on master. For this case, I'd be happy to do it on your behalf.

**Comment by zacharied** [comment]
If you wouldn't mind, I'd appreciate that. I could use an example for future reference, too.

**Comment by zacharied** [comment]
Also, I incidentally wrote a [script](https://gist.github.com/zacharied/00c55282e3a48ccc19603bb2aaca0bc2) earlier that does what you suggested!

**Comment by jwilm** [comment]
@zacharied thought you might like to know that I've got a [branch] that sets `_NET_WM_PID`. You might care about this because it allows supports what your script does without relying on the window title. See the commit messages for details.

[branch]: https://github.com/jwilm/alacritty/tree/set-net-wm-pid

**Pull request closed without merging**